node/src/blockchain/blockchain_bridge.rs

```diff
@@ -1140,22 +1404,35 @@ mod tests {
             }
         );
         let scan_error_msg = accountant_recording.get_record::<ScanError>(1);
-        assert_eq!(*scan_error_msg, ScanError {
-            scan_type: ScanType::PendingPayables,
-            response_skeleton_opt: Some(ResponseSkeleton { client_id: 1234, context_id: 4321 }),
-            msg: "Aborting scanning; request of a transaction receipt \
-         for '0x000000000000000000000000000000000000000000000000000000000001348d' failed due to 'QueryFailed(\"bad bad bad\")'".to_string()
-        });
-        TestLogHandler::new().exists_log_containing("WARN: BlockchainBridge: Aborting scanning; request of a transaction receipt \
-         for '0x000000000000000000000000000000000000000000000000000000000001348d' failed due to 'QueryFailed(\"bad bad bad\")'");
+        assert_eq!(
+            *scan_error_msg,
+            ScanError {
+                scan_type: ScanType::PendingPayables,
+                response_skeleton_opt: Some(ResponseSkeleton {
+                    client_id: 1234,
+                    context_id: 4321
+                }),
+                msg: "Aborting scanning; 1 transactions succeed and 3 transactions failed"
+                    .to_string()
+            }
+        );
+        TestLogHandler::new().exists_log_containing("WARN: BlockchainBridge: Aborting scanning; 1 transactions succeed and 3 transactions failed");
     }
 
     #[test]
     fn blockchain_bridge_can_return_report_transaction_receipts_with_an_empty_vector() {
```

Comment:
This test is probably a nonsense. (You're of no guiltiness)

This request would never be constructed or shouldn't be at least.
We maybe should look at the code in the Scanners whether we terminate the scan if we find no
pending payables in the database table. If we do, we should not bother sending a message to the BlockchainBridge. Okay, I checked on it and it was right. That means we're testing an unreal condition. 
Such a test doesn't deserve a place in the test suit or the codebase in overall. 
 
```
        let msg = RequestTransactionReceipts {
            pending_payable: vec![],
            response_skeleton_opt: None,
        };
```

Reply:
**URGENCY STATUS: LOW, BUT I'D BE SO KEEN TO DO THIS AS I HATE MAINTAINING GARBAGE IN THE CODEBASE." In case you think it's risky, go and check my words. 

Actually, digging deeper in this problematics, if you want to silent all your fears, you either can implement an immediate return in case the an empty vector was just received, or you can implement even a panic as, obviously, it would point to a bug, and we want to know if we did a floppy job with some consideration and the empty vector truly comes along.

Reply:
We believe this test should remain because it will help a developer to understand BlockchainBridge's behaviour regarding an empty request.
It also tests this corner case.

please let us know your thoughts :)

Reply:
If you ask what I would've done then the answer is "I'd never allowed to have this test, and would remove it in this other case". 

I offer an alternative to what I'm saying and which will also be along your thoughts: if you want to teach the developer of how these processes work you could change it into a panic. If we receive an empty vector, blow up the Node, as it's not expected. We shouldn't do this wasteful job and send a msg with an empty vector just to have BlockchainBridge to conclude it's empty and return another actor message. Prevent it, and punish if it somehow still happen.